Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -2697,15 +2696,71 @@ class HTTPClientTests: XCTestCase {
             XCTAssertNoThrow(try server.close().wait())
         }
 
-        let request = try Request(url: "https://localhost:\(server.localAddress!.port!)", method: .GET)
-        let task = self.defaultClient.execute(request: request, delegate: TestHTTPDelegate())
+        // We set the connect timeout down very low here because on NIOTS this manifests as a connect
+        // timeout.
+        let config = HTTPClient.Configuration(timeout: HTTPClient.Configuration.Timeout(connect: .milliseconds(100), read: nil))
+        let client = HTTPClient(eventLoopGroupProvider: .shared(self.clientGroup), configuration: config)
+        defer {
+            XCTAssertNoThrow(try client.syncShutdown())
+        }
+
+        let request = try Request(url: "https://127.0.0.1:\(server.localAddress!.port!)", method: .GET)
+        let task = client.execute(request: request, delegate: TestHTTPDelegate())
+
+        XCTAssertThrowsError(try task.wait()) { error in
+            #if os(Linux)
+                XCTAssertEqual(error as? NIOSSLError, NIOSSLError.uncleanShutdown)
+            #else
+                if isTestingNIOTS() {
+                    XCTAssertEqual(error as? ChannelError, .connectTimeout(.milliseconds(100)))
+                } else {
+                    XCTAssertEqual((error as? IOError).map { $0.errnoCode }, 54)
```

Comment:
any reason to use `54` over `ECONNRESET`?

Reply:
I just cribbed it from the test above, happy to swap it.